### PR TITLE
Fix a bug on validation for MAC address

### PR DESCRIPTION
**Pull request by tiwanari**
I found a small but fatal bug. 

The validation of a MAC address is wrong. If the string is not set, `""` will be returned not null.

### CODE REVIEW

**Review by mandaiy — COMMENTED** [review]
At a glance, we have a similar validation on L138.
Doesn't the validation work correctly too?

According to the README of KVS-schema, we can specify default values in schema classes.
The README seems not to tell what the default value of string is if we don't provide a default value explicitly (and the default value is actually an empty string, right?).
So I think we should add the default value for the address explicitly (empty string seems appropriate).
Could you add the default value for the string?

**Comment by tiwanari** [comment]
> Doesn't the validation work correctly too?

No, thank you for pointing out that!

Your suggestion is quite reasonable. Will do:)

**Comment by tiwanari** [comment]
I did:)

**Comment by mandaiy** [comment]
Ah, I meant that why not add the default value to the definition of MachinePreference like
```
@Table(name = "machine_preferences")
public class MachinePreferencesSchema {
    // ...
    @Key(name = "bluetooth_address")
    String macAddress = "";
    // ...
```

But calling ``getMacAddress()`` with providing a default argument also makes sense and might be enough.
It's up to you whether adding the default value to the class definition or not.

**Review by mandaiy — APPROVED** [review]

**Comment by tiwanari** [comment]
Ah, sorry for my misunderstanding.. Hmm, in this case, both of them are reasonable but this kind of bugs is easily made and confusing. I guess this is a good time to start using guava which has useful methods like `isNullOrEmpty` . What do you think?

**Comment by tiwanari** [comment]
I think it's natural the method returns `null` (that's why the bug was made ) and returning `""` as well (in the Java world). So, using the until method, we don't have to specify the default value from my viewpoint.

File: app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivity.java [hunk 1]
```diff
@@ -134,9 +134,10 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     private void setUpToolbar() {
         // get device address to show it on toolbar
-        String deviceAddress = MachinePreferences.get(getApplicationContext()).getMacAddress();
+        String deviceAddress = MachinePreferences.get(getApplicationContext()).getMacAddress("");
```
Comment (mandaiy) [inline app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivity.java]: If I were you I'd write
```java
String defaultString = getResources().getString(R.string.programming_noTargetDevice);
String deviceAddress = MachinePreferences.get(getApplicationContext()).getMacAddress(defaultString);
```

**Comment by mandaiy** [comment]
Introducing guava is fine to me.
But I think there are a lot of lines we can refactor by using the library and I want to replace such lines at a time.

So can you close this pull request?
And let's discuss the library and list up the lines in another issue or pull request.

File: app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivity.java (hunk 1 above)
Comment (tiwanari) [reply app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivity.java]: looks good. thanks!

**Comment by tiwanari** [comment]
okay, after modifying lines you mentioned, I'll close this and raise an issue

**Pull request merged**